### PR TITLE
OTLP Exporter: Integration test improvements

### PR DESCRIPTION
**Pull request by CodeBlanch**
Related to #3092

Updated the OTLP integration tests to verify default batch/periodic modes in addition to force flush.

Also fixed a bug I noticed where `null` `BatchExportProcessorOptions` would blow up the OTLP tracing builder extension.

### CODE REVIEW

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/IntegrationTests.cs [hunk 1]
```diff
@@ -58,11 +61,16 @@ public void TraceExportResultIsSuccess(OtlpExportProtocol protocol, string endpo
...
 #endif
+            using EventWaitHandle handle = new ManualResetEvent(false);
 
             var exporterOptions = new OtlpExporterOptions
             {
                 Endpoint = new Uri($"http://{CollectorHostname}{endpoint}"),
                 Protocol = protocol,
+                BatchExportProcessorOptions = new()
+                {
+                    ScheduledDelayMilliseconds = 10000,
```
Comment (reyang) [inline test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/IntegrationTests.cs:72]: minor: given this is used later, maybe use a local const?

**Review by reyang — APPROVED** [review]
LGTM.

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/IntegrationTests.cs (hunk 1 above)
Comment (CodeBlanch) [reply test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/IntegrationTests.cs]: Updated

**Review by alanwest — APPROVED** [review]
OTLP (oh this looks pretty)

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/IntegrationTests.cs [hunk 2]
```diff
@@ -90,16 +99,27 @@ public void TraceExportResultIsSuccess(OtlpExportProtocol protocol, string endpo
             activity?.Stop();
 
             Assert.NotNull(delegatingExporter);
-            Assert.True(tracerProvider.ForceFlush());
+
+            if (forceFlush)
+            {
+                Assert.True(tracerProvider.ForceFlush());
+            }
+            else if (exporterOptions.ExportProcessorType == ExportProcessorType.Batch)
```
Comment (pellared) [inline test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/IntegrationTests.cs:107]: Do you plan to test `ExportProcessorType.Simple` as well?

**Review by pellared — APPROVED** [review]

**Comment by cijothomas** [comment]
> OTLP (oh this looks pretty)

Add here (https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/glossary.md) and make it official 😄

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/IntegrationTests.cs (hunk 2 above)
Comment (CodeBlanch) [reply test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/IntegrationTests.cs]: Originally I had planned to just cover the default options but it does seem good to have this. Added.

**Review by cijothomas — APPROVED** [review]

**Pull request merged**